### PR TITLE
Simplify geolocation-API/watchPosition_permission_deny.https.html

### PR DESCRIPTION
**Pull request by marcoscaceres**
Modernized the tests. Happy in Chrome ✅.

### CODE REVIEW

File: geolocation-API/watchPosition_permission_deny.https.html [hunk 1]
```diff
@@ -1,33 +1,45 @@
...
 
-// Rewrite http://dev.w3.org/geo/api/test-suite/t.html?00151
-test(function() {
-  assert_greater_than_equal(id, -2147483648);
-  assert_less_than_equal(id, 2147483647);
-}, 'Check that watchPosition returns a long');
+  promise_test(async (t) => {
+    t.add_cleanup(resetPermission);
+    await test_driver.set_permission({ name: "geolocation" }, "denied");
+    const promise = new Promise((resolve, reject) => {
```
Comment (rakuco) [inline geolocation-API/watchPosition_permission_deny.https.html:36]: Isn't it cleaner to set `result` here directly?

File: geolocation-API/watchPosition_permission_deny.https.html [hunk 2]
```diff
@@ -1,33 +1,45 @@
...
+      navigator.geolocation.watchPosition(reject, resolve);
+    });
+    const result = await promise;
+    assert_true(
```
Comment (rakuco) [inline geolocation-API/watchPosition_permission_deny.https.html:40]: You could just add this check to the test above, but I understand it if you prefer to keep the two tests separate.

**Review by rakuco — APPROVED** [review]
lgtm modulo the general question about the need for `add_cleanup`.

File: geolocation-API/watchPosition_permission_deny.https.html [hunk 3]
```diff
@@ -1,33 +1,45 @@
...
+    const errorPromise = new Promise((resolve, reject) => {
+      id = navigator.geolocation.watchPosition(reject, () => {
+        errorCBCalled = true;
+        resolve();
+      });
+    });
+    assert_false(
+      errorCBCalled,
+      "error callback must not be called synchronously"
+    );
```
Comment (reillyeon) [inline geolocation-API/watchPosition_permission_deny.https.html:28]: I notice there isn't a similar check in the getCurrentPosition or permissions policy tests I just reviewed. Making sure we have full coverage would be good.

**Review by reillyeon — APPROVED** [review]

**Comment by marcoscaceres** [comment]
I'm going to take this as is, but then I think I'm going to merge these permission tests into a single file. That way, we can make sure they are consistent.

**Pull request merged**